test/spec/lifecycle-test.js

```diff
@@ -456,6 +458,405 @@ describe("Lifecycle tests", function () {
+        const doc = node ? node.ownerDocument : document;
+        const evt = doc.createEvent("MouseEvents");
+        evt.initMouseEvent(
+          "mousemove",
+          true,
+          true,
+          window,
+          0,
+          0,
+          0,
+          x,
+          y,
+          false,
+          false,
+          false,
+          false,
+          0,
+          null
+        );
+        doc.dispatchEvent(evt);
+      };
+
+      const getCurrentPosition = wrapper => {
+        const { x, y, w, h } = wrapper.props();
+        return calcGridItemPosition(
+          wrapper.instance().getPositionParams(),
+          x,
+          y,
+          w,
+          h,
+          wrapper.state()
+        );
+      };
+
+      const resizeTo = (wrapper, preventMouseUp, currentPosition, x, y) => {
+        const node = wrapper.getDOMNode();
+        TestUtils.Simulate.mouseDown(node, {
+          clientX: currentPosition.left,
+          clientY: currentPosition.top
+        });
+        mouseMove(node, x, y);
+
+        // In some test cases we want to take measurements before mouseUp occurs
+        if (!preventMouseUp) {
+          TestUtils.Simulate.mouseUp(node);
+        }
+      };
+
+      const findGridItemByText = (wrapper, id) =>
+        wrapper.findWhere(node => {
+          const isGridItem = node.instance() instanceof GridItem;
+          const hasSpecifiedText = node.find({ children: id }).exists();
+
+          return isGridItem && hasSpecifiedText;
+        });
+
+      const findHandleForGridItem = (wrapper, handle) =>
+        wrapper.find(`.react-resizable-handle-${handle}`);
+
+      const getGridItemData = (wrapper, id) =>
+        wrapper.state("layout").find(item => item.i == id);
+
+      beforeEach(() => {
+        const wrapper = mount(<ResizableLayout />);
+        gridLayout = wrapper.find("ReactGridLayout");
+      });
+
+      it("sets up resizable handles", () => {
+        const gridItem0 = findGridItemByText(gridLayout, 0);
+        // Ensure every handle is present on the target element
+        ["n", "ne", "e", "se", "s", "sw", "w", "nw"].forEach(handle => {
+          expect(findHandleForGridItem(gridItem0, handle).exists()).toEqual(
+            true
+          );
+        });
+      });
+
+      it("resizes from n handle", () => {
+        const itemId = 6;
+        const gridItem6 = findGridItemByText(gridLayout, itemId);
+        const handleElement = findHandleForGridItem(gridItem6, "n");
+        const pos = getCurrentPosition(gridItem6);
+
+        const positionBeforeResize = getGridItemData(gridLayout, itemId);
+        // Resize up two rows
+        resizeTo(handleElement, false, pos, pos.left, pos.top - rowHeight * 2);
+        const positionAfterResize = getGridItemData(gridLayout, itemId);
+        expect(positionAfterResize).toEqual({
+          ...positionBeforeResize,
+          h: positionBeforeResize.h + 2,
+          y: positionBeforeResize.y - 2
+        });
+      });
```

Comment:
I also added some generic `resize by handle X` tests as well since I couldn't find any that currently existed.